src/pages/fieldManagement/settings/DefaultFieldTable.jsx

```diff
@@ -114,22 +114,57 @@ export default function DefaultFieldTable({
             setFormSettings(getInitialFormState(siteSettings));
             onCloseEditor();
           }}
-          onSubmit={() => {
-            if (editField.id === 'region') {
-              putSiteSetting(
+          onSubmit={async () => {
+            if (editField?.id === 'region') {
+              const success = await putSiteSetting(
                 editField.backendPath,
                 formSettings.regions,
-              ).then(success => {
-                if (success) onCloseEditor();
-              });
+              );
+              if (success) onCloseEditor();
             }
-            if (editField.id === 'species') {
-              putSiteSetting(
+            if (editField?.id === 'species') {
+              const success = await putSiteSetting(
                 editField.backendPath,
                 formSettings.species,
-              ).then(success => {
-                if (success) onCloseEditor();
-              });
+              );
+              if (success) onCloseEditor();
+            }
+            if (editField?.id === 'relationship') {
+              const newSettings = reduce(
+                formSettings.relationships,
+                (memo, currentRelationships, key) => {
+                  const dedupedRelationships = currentRelationships.filter(
+                    (entry, index) => {
+                      return (
+                        currentRelationships.indexOf(entry) === index
+                      );
+                    },
+                  );
+                  if (
+                    currentRelationships.length >
+                    dedupedRelationships.length
+                  ) {
+                    alert(
+                      intl.formatMessage({
+                        id:
+                          'DUPLICATE_RELATIONSHIPS_FOUND_AND_REMOVED',
+                      }),
+                    );
+                  }
+                  let newRelationshipChunk = {};
+                  newRelationshipChunk[key] = dedupedRelationships;
+                  return { ...memo, ...newRelationshipChunk };
```

Comment:
```
return {
  ...memo,
  [key]: dedupedRelationships,
}
```

Reply:
Done. Good eye.

Reply:
TIL that you can add a new key-value pair to an object without a string using [someVariable] as the key. Awesome! Very good to know this.